sites/partners/lib/formatApplicationData.ts

```diff
@@ -232,10 +288,42 @@ export const parseApplicationData = (applicationData: ApplicationUpdate) => {
 
   const phoneNumber = applicationData.applicant.phoneNumber
 
+  const preferences = (() => {
```

Comment:
@dominikx96 Is there a way this block of code could be expressed in a more English-like readable manner, or at the very least provide some comments? I'd have to read this over a number of times before generally understanding what it's trying to accomplish.

Reply:
@jaredcwhite agree! I refactored function names to be more intuitive, renamed variables, and added some comments. I think will be easier when we'' have backend types for this, I temporary typed it as:

`const preferencesApiData = applicationData.preferences as Record<string, any>`

because `object` is not a valid type. Talked with @pbn4 and we should change it after production data migration to a proper format.

Hope the current version of the code is more readable :)